napari/_qt/tests/test_qt_viewer.py

```diff
@@ -22,136 +23,31 @@ def test_qt_viewer(qtbot):
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.random((10, 15, 20))
-    viewer.add_image(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 3
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_pyramid(qtbot):
-    """Test adding image pyramid."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    shapes = [(40, 20), (20, 10), (10, 5)]
-    np.random.seed(0)
-    data = [np.random.random(s) for s in shapes]
-    viewer.add_image(data, is_pyramid=True)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_labels(qtbot):
-    """Test adding labels image."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.randint(20, size=(10, 15))
-    viewer.add_labels(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_points(qtbot):
-    """Test adding points."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = 20 * np.random.random((10, 2))
-    viewer.add_points(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_vectors(qtbot):
-    """Test adding vectors."""
-    viewer = ViewerModel()
+@pytest.mark.parametrize('layer_type, data, ndim', [
+    (LayerTypes.IMAGE, np.random.random((10, 15)), 2),
+    (LayerTypes.VOLUME, np.random.random((10, 15, 20)), 3),
+    (LayerTypes.PYRAMID, [np.random.random(s) for s in [(40, 20), (20, 10), (10, 5)]], 2),
+    (LayerTypes.LABELS, np.random.randint(20, size=(10, 15)), 2),
+    (LayerTypes.POINTS, 20 * np.random.random((10, 2)), 2),
+    (LayerTypes.VECTORS, 20 * np.random.random((10, 2, 2)), 2),
+    (LayerTypes.SHAPES, 20 * np.random.random((10, 4, 2)), 2),
+    (LayerTypes.SURFACE, (np.random.random((10, 3)), np.random.randint(10, size=(6, 3)),
+                 np.random.random(10)), 3)
+])
+def test_add_layer(qtbot, layer_type, data, ndim):
```

Comment:
the stuff that gets checked here can also be identical to the stuff that gets checked in the tests in `test_viewer.py` and `test_napari.py` the only difference is now
```python
viewer = ViewerModel()
view = QtViewer(viewer)
```
and the use of `view` in the tests. We can do the same tricks with `viewer.dims.ndisplay` going from 2 -> 3 -> 2 as I describe in some of my comments below